pkg/ebpf/encoding.go

```diff
@@ -0,0 +1,180 @@
+package ebpf
```

Comment:
I think it would make sense to move this to a separate `encoding` or `serialization` package. Do you agree?